### PR TITLE
Get-DbaXEventSessionTarget and test

### PR DESCRIPTION
**Pull request by potatoqualitee**

### CODE REVIEW

File: functions/Get-DbaXEventSessionTarget.ps1 [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+		[string[]]$Session,
+		[string[]]$Target,
+		[parameter(ValueFromPipeline, ParameterSetName = "piped", Mandatory)]
+		[Microsoft.SqlServer.Management.XEvent.Session[]]$SessionObject,
+		[switch]$Silent
+	)
+	
+	begin {
+		if ([System.Reflection.Assembly]::LoadWithPartialName("Microsoft.SqlServer.Management.XEvent") -eq $null) {
+			Stop-Function -Message "SMO version is too old. To collect Extended Events, you must have SQL Server Management Studio 2012 or higher installed."
```
Comment (FriedrichWeinmann) [inline functions/Get-DbaXEventSessionTarget.ps1]: should have a return on the next line.
Not a technical requirement for this function, but it is neater and avoids errors if someone later adds things to the begin block

File: functions/Get-DbaXEventSessionTarget.ps1 [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		}
+		
+		function get-target ($Sessions, $server) {
```
Comment (FriedrichWeinmann) [inline functions/Get-DbaXEventSessionTarget.ps1]: I know it's an internal function, but still, it should have a `[CmdletBinding()]` (and thus a param block) anyway
Also watch your casing (it's upper-case when you later call it)

File: functions/Get-DbaXEventSessionTarget.ps1 [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+			foreach ($xsession in $Sessions) {
+				
+				if ($null -eq $server) {
+					$server = $xsession.Parent
+				}
+				
+				if ($Session -and $xsession.Name -notin $Session) { continue }
```
Comment (FriedrichWeinmann) [inline functions/Get-DbaXEventSessionTarget.ps1]: Scope boundary violation detected.

File: functions/Get-DbaXEventSessionTarget.ps1 (hunk 2 above)
Comment (potatoqualitee) [reply functions/Get-DbaXEventSessionTarget.ps1]: fixed

File: functions/Get-DbaXEventSessionTarget.ps1 (hunk 1 above)
Comment (potatoqualitee) [reply functions/Get-DbaXEventSessionTarget.ps1]: fixed

File: functions/Get-DbaXEventSessionTarget.ps1 (hunk 3 above)
Comment (potatoqualitee) [reply functions/Get-DbaXEventSessionTarget.ps1]: wut how to fix

Comment (potatoqualitee) [reply functions/Get-DbaXEventSessionTarget.ps1]: nm i see

Comment (FriedrichWeinmann) [reply functions/Get-DbaXEventSessionTarget.ps1]: By adding a `$Session` parameter to `Get-Target` and passing the original `$Session` parameter of `Get-DbaXEventSessionTarget` though to it.
Basically:
 - Add `$Session`-Parameter to param block of  `Get-Target`
 - Append `-Session $Session` to all calls of Get-Target later in the function

Comment (potatoqualitee) [reply functions/Get-DbaXEventSessionTarget.ps1]: AHHHHHH thank you 🙇 I also added Target

**Pull request merged**